sympy/tensor/array/expressions/array_expressions.py

```diff
@@ -257,7 +274,14 @@ def _normalize(self):
             diagonal_indices = [tuple(cumulative_ranks2[i] + k for k in j) for i, arg in diagonals.items() for j in arg.diagonal_indices]
             return _permute_dims(_array_diagonal(tp, *diagonal_indices), _af_invert(inverse_permutation))
 
-        return self.func(*args, normalize=False)
+        return self.func(*args, canonicalize=False)
+
+    def doit(self, **kwargs):
+        deep = kwargs.get("deep", True)
```

Comment:
Just a quick check, should the default behaviour be that the canonicalization is performed recursively?

Reply:
Yes, this is compatible with the default behaviour of `.doit()` in SymPy.